packages/tasit-action/src/TasitAct.js

```diff
@@ -0,0 +1,144 @@
+import "ethers/dist/shims.js";
+// Note: ethers SHOULD be imported from their main object
+// shims aren't injected with package import
+import { ethers } from "ethers";
+
+class Utils {
+  static isAddress = address => {
+    return typeof address === "string" && address.match(/^0x[0-9A-Fa-f]{40}$/);
+  };
+
+  static isABI = abi => {
+    return abi && Array.isArray(abi);
+  };
+
+  // https://github.com/ethers-io/ethers.js/blob/db383a3121bb8cf5c80c5488e853101d8c1df353/src.ts/utils/properties.ts#L20
+  static isEthersJsSigner = signer => {
+    return signer && signer._ethersType === "Signer";
+  };
+}
+
+class Subscription {
+  #txPromise;
+  #provider;
+
+  constructor(txPromise, provider) {
+    this.#txPromise = txPromise;
+    this.#provider = provider;
+  }
+
+  on = async (trigger, callback) => {
+    let blockHeight = 0;
+
+    if (trigger === "confirmation") blockHeight = 1;
+    else {
```

Comment:
Consider erroring out as early as possible in functions, and then continuing execution onwards as non-error case.

Reply:
```
if (trigger !== "confirmation") {
  throw new Error(`Invalid subscription trigger, use: ["confirmation"]`);
}
blockHeight = 1;
```

Reply:
I find that this forces the developer to consider "non-happy-paths" early and often